src/OptionSettlementEngine.sol

```diff
@@ -608,6 +616,15 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
             uint16 bucketIndex = unexercisedBucketIndices[unexercisedBucketsIndex];
             Bucket storage claimBucketInfo = claimBucketArray[bucketIndex];
 
+            // skip bucket if it happens to be in the current window
+            if (_getPeriodBucket() == claimBucketInfo.periodsAfterEpoch) {
+                if (unexercisedBucketIndices.length == 1) {
```

Comment:
this can happen if a lot of writes happen in the current window with no previous buckets available for exercise. there exist plenty of options holders, but no assignable options.